src/app/modules/item/pages/item-details/item-details.component.ts

```diff
@@ -7,17 +7,20 @@ import { LayoutService } from '../../../../shared/services/layout.service';
 import { RouterLinkActive } from '@angular/router';
 import { TaskTab } from '../item-display/item-display.component';
 import { Mode, ModeService } from 'src/app/shared/services/mode.service';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { fromEvent, Observable, of, Subject } from 'rxjs';
+import { catchError, endWith, last, map, shareReplay, switchMap, take, takeUntil } from 'rxjs/operators';
 import { ConfigureTaskOptions } from '../../services/item-task.service';
+import { BeforeUnloadComponent } from 'src/app/shared/guards/before-unload-guard';
+import { ItemContentComponent } from '../item-content/item-content.component';
 
 @Component({
   selector: 'alg-item-details',
   templateUrl: './item-details.component.html',
   styleUrls: [ './item-details.component.scss' ],
 })
-export class ItemDetailsComponent implements OnDestroy {
+export class ItemDetailsComponent implements OnDestroy, BeforeUnloadComponent {
   @ViewChild('progressTab') progressTab?: RouterLinkActive;
+  @ViewChild(ItemContentComponent) itemContentComponent?: ItemContentComponent;
```

Comment:
Isn't "item content" the parent of "item details"? A component should not have a reference to its parent!  (and in a `ViewChild` ? 🤨) Anyway item content has also a viewchild reference to item details... very smelly.

Reply:
item content is the child of item details, the structure is `item-details` > `item-content` > `item-display`

Another way around to retrieve that `saveAnswerAndState()` function in item-details would be to bubble it up from item-display through an event, but that would mean an `EventEmitter<() => Observable<void>>`, smelly as well…

Reply:
Ok my bad, I must have mixed details and display.

Reply:
I could also create a service to which item-task-answer.service could register the `saveAnswerAndState()` function and item-details.component could retrieve this function from the service
To me it's smelly too :confused: